File: src/main/java/com/github/sorusclient/client/module/ModuleDisableable.java
package com.github.sorusclient.client.module;

import com.github.sorusclient.client.setting.Setting;
import com.github.sorusclient.client.setting.ConfigurableData;

import java.util.List;

public class ModuleDisableable extends Module {

    private final Setting<Boolean> enabled;

    public ModuleDisableable(String id) {
        super(id);

        this.register("enabled", this.enabled = new Setting<>(false));
    }

    public void setEnabled(boolean enabled) {
        this.enabled.setValue(enabled);
    }

    public boolean isEnabled() {
        return this.enabled.getValue();
    }

    @Override
    public void addSettings(List<ConfigurableData> settings) {
        super.addSettings(settings);
        settings.add(new ConfigurableData.Toggle("Enabled", this.enabled));
    }

}

File: src/main/kotlin/com/github/sorusclient/client/module/ModuleDisableable.kt
package com.github.sorusclient.client.module

import com.github.sorusclient.client.setting.DisplayedSetting
import com.github.sorusclient.client.setting.DisplayedSetting.Toggle
import com.github.sorusclient.client.setting.Setting

open class ModuleDisableable(id: String) : Module(id) {

    private var enabled: Setting<Boolean>

    init {
        register("enabled", Setting(false).also { enabled = it })
    }

    fun setEnabled(enabled: Boolean) {
        this.enabled.realValue = enabled
    }

    fun isEnabled(): Boolean {
        return enabled.value
    }

    override fun addSettings(settings: MutableList<DisplayedSetting>) {
        super.addSettings(settings)
        settings.add(Toggle("Enabled", enabled))
    }

}